### PR TITLE
Pass a real route object into the controller

### PR DESCRIPTION
**Pull request by jneen**
This allows us to package up the routing information into a single object that is passed into the controller. I've also realized the RoutingTable object, which greatly simplifies the navigation through the routes list.

### CODE REVIEW

**Comment by timarney** [comment]
This is great.  Few comments and the linting errors to cleanup.

Outside of that please bump the version in package.json + track the changes in the changelog file

Added, Updated, Removed.

@maxneuvians , @dsamojlenko feedback would be appreciated.

**Comment by maxneuvians** [comment]
LGTM

File: app.js [hunk 1]
```diff
@@ -88,7 +87,9 @@ app.locals.hasData = hasData
 app.locals.basedir = path.join(__dirname, './views')
 app.set('views', [path.join(__dirname, './views')])
 
-configRoutes(app, routes)
+app.routes = {
+  default: configRoutes(app, routes)
```
[the remark this answers is not in the document]
Comment (jneen) [reply app.js]: In #58 this object will change to a `LocaleGroup`, which will have `en` and `fr` keys - giving two separate routing tables for each locale.

[the remark this answers is not in the document]
Comment (jneen) [reply app.js]: The reason I used `default` is in the case that the user wanted to use more than one routing table, though I'm not sure why that would be the case - it seemed to be supported by the functionality in the previous routes helper file...

File: utils/route.helpers.js [hunk 2]
```diff
@@ -1,151 +1,91 @@
...
- * @param {Array} routes array of route objects { name: "start", path: "/start" },
- * @returns { name: "", path: "" }
- */
-const getPreviousRoute = (name, routes = defaultRoutes) => {
-  const route = getRouteWithIndexByName(name, routes)
-
-  if (!routeHasIndex(route) && process.env.NODE_ENV !== 'production') {
-    throw new Error(`Previous route error can't find => "${name}"`)
+class Route {
+  constructor(table, index, conf) {
```
[the remark this answers is not in the document]
Comment (jneen) [reply utils/route.helpers.js]: I see! The `table` is a back-reference to the `RoutingTable` instance that this route is a part of. `conf` is literally the object in the user's configuration that has `name` and `path` keys, but can also have other keys that will be copied here.

I can add comments - now that this is merged where's the best place to put those?

File: utils/route.helpers.spec.js [hunk 3]
```diff
@@ -1,72 +1,47 @@
...
-  routeHasIndex,
 } = require('./index')
 
-const testRoutes = [
+const testRoutes = makeRoutingTable([
   { name: 'start', path: '/start' },
   { name: 'personal', path: '/personal' },
   { name: 'confirmation', path: '/confirmation' },
-]
+], { directory: '/tmp' })
```
[the remark this answers is not in the document]
Comment (jneen) [reply utils/route.helpers.spec.js]: it's the directory to load routes from, by default `./routes` from the base of the app. This test doesn't actually load any of the routes, so it's not neede.

File: utils/route.helpers.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (timarney) [reply utils/route.helpers.js]: Let's just add them above the constructor i.e. 

Just something that easy to `glance at` vs having to track down the calling code to see what's being passed in :)

File: app.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (timarney) [reply app.js]: If we can I would just remove the default.  The old was more of a does a user defined config exist?  If not use the default.

If we end up needing more than once routing table later we can deal with it at that point.

**Pull request merged**